CONTRIBUTING.md

```diff
@@ -17,13 +17,314 @@
+    ```
+   For an exhaustive list of tasks, use:
+    ```
+    $ ./gradlew tasks --all
+    ```
+
+6. Make sure you can build and run tests.
+
+   Since Beam is a large project, usually, you will want to limit testing to the particular module you are working on. Gradle will build just the necessary things to run those tests. For example:
+    ```
+    $ ./gradlew -p sdks/go check
+    $ ./gradlew -p sdks/java/io/cassandra check
+    $ ./gradlew -p runners/flink check
+    ```
+
+7. Now you may want to set up your preferred IDE and other aspects of your development
+   environment. See the Developers' wiki for tips, guides, and FAQs on:
+  - [IntelliJ](https://cwiki.apache.org/confluence/display/BEAM/Using+IntelliJ+IDE)
+  - [Java](https://cwiki.apache.org/confluence/display/BEAM/Java+Tips)
+  - [Python](https://cwiki.apache.org/confluence/display/BEAM/Python+Tips)
+  - [Go](https://cwiki.apache.org/confluence/display/BEAM/Go+Tips)
+  - [Website](https://cwiki.apache.org/confluence/display/BEAM/Website+Tips)
+  - [Gradle](https://cwiki.apache.org/confluence/display/BEAM/Gradle+Tips)
+  - [Jenkins](https://cwiki.apache.org/confluence/display/BEAM/Jenkins+Tips)
+  - [FAQ](https://cwiki.apache.org/confluence/display/BEAM/Contributor+FAQ)
+
+### Create a Pull Request
+
+1. Make your code change. Every source file needs to include the Apache license header. Every new dependency needs to
+   have an open source license [compatible](https://www.apache.org/legal/resolved.html#criteria) with Apache.
+
+2. Add unit tests for your change.
+
+3. Use descriptive commit messages that make it easy to identify changes and provide a clear history.
+
+4. When your change is ready to be reviewed and merged, create a pull request.
+
+5. Link to the issue you are addressing in your pull request.
+
+6. The pull request and any changes pushed to it will trigger [pre-commit
+   jobs](https://cwiki.apache.org/confluence/display/BEAM/Contribution+Testing+Guide#ContributionTestingGuide-Pre-commit). If a test fails and appears unrelated to your
+   change, you can cause tests to be re-run by adding a single line comment on your
+   PR:
+    ```
+    retest this please
+    ```
+Pull request template has a link to a [catalog of trigger phrases](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md)
+that start various post-commit tests suites. Use these sparingly because post-commit tests consume shared development resources.
+
+### Review Process and Releases
+
+#### Get Reviewed
+
+Your pull requests should automatically have reviewers assigned within a few hours of opening it.
+If that doesn't happen for some reason, you can also request a review yourself.
+
+1. Pull requests can only be merged by a
+   [Beam committer](https://home.apache.org/phonebook.html?pmc=beam).
+   To find a committer for your area, either:
+  - look for similar code merges, or
+  - ask on [dev@beam.apache.org](/community/contact-us/)
```

Comment:
should this be `https://beam.apache.org/community/contact-us/`?

Reply:
looks like other links might be broken.

Reply:
Good call, I didn't notice these when I migrated some content from the website. Thanks!